# Activities


## Activity 2.8
$\begin{pmatrix} 1 & 1 & 1 & 3 \\ 0 & 1 & 1 & 2 \\ 0 & 0 & 1 & 2 \end{pmatrix}$

This last matrix is in *row echelon form*, or simply *echelon form*.

**Definition (Row echelon form)** A matrix is said to be in *row echelon form* (or *echelon form*) if it has the following properties:

1) Every non-zero row begins with a leading one. 

2) A leading one in a lower row is further to the right.

3) Zero rows are at the bottom of the matrix.

Check that the above matrix satisfies these properties.

The above matrix satifies all the properties.

**Definition (Reduced row echelon form)** A matrix is said to be in *reduced row echelon form* (or *reduced echelon form*) if it has the following four properties:

1) Every non-zero row begins with a leading one. 

2) A leading one in a lower row is further to the right.

3) Zero rows are at the bottom of the matrix.

4) Every column with a leading one has zeros elsewhere.

## Activity 2.10

Do this: check that $\begin{bmatrix} 1 & 1 & 1 \\ 2 & 1 & 1 \\ 1 & -1 & 2 \end{bmatrix} \begin{bmatrix} 1 \\ 0 \\ 2 \end{bmatrix}=\begin{bmatrix} 3 \\ 4 \\ 5 \end{bmatrix}$.

First we denote $Ax$ as an augmented matrix $A$:

$\left[\begin{array}{ccc|c}1 & 1 & 1 & 1 \\2 & 1 & 1 & 0 \\ 1 & -1 & 2 & 2 \\\end{array}\right]$

We now use Gauss Elimination to see if $Ax=b$.

$\left[\begin{array}{ccc|c}1 & 1 & 1 & 1 \\2 & 1 & 1 & 0 \\ 1 & -1 & 2 & 2 \\\end{array}\right]$

$\Updownarrow$

Subtract $2R_1$ from $R_2$.

$\left[\begin{array}{ccc|c}1 & 1 & 1 & 1 \\0 & -1 & -1 & -2 \\ 1 & -1 & 2 & 2 \\\end{array}\right]$

$\Updownarrow$

Subtract $1R_1$ from $R_3$.

$\left[\begin{array}{ccc|c}1 & 1 & 1 & 1 \\0 & -1 & -1 & -2 \\ 0 & -2 & 1 & 1 \\\end{array}\right]$

$\Updownarrow$

Multiply $R_2$ with $-1$.

$\left[\begin{array}{ccc|c}1 & 1 & 1 & 1 \\0 & 1 & 1 & 2 \\ 0 & -2 & 1 & 1 \\\end{array}\right]$

$\Updownarrow$

Add $2R_2$ to $R_3$

$\left[\begin{array}{ccc|c}1 & 1 & 1 & 1 \\0 & 1 & 1 & 2 \\ 0 & 0 & 3 & 5 \\\end{array}\right]$

$\Updownarrow$

Multiply $R_3$ with $\frac{1}{3}$

$\left[\begin{array}{ccc|c}1 & 1 & 1 & 1 \\0 & 1 & 1 & 2 \\ 0 & 0 & 1 & \frac{5}{3} \\\end{array}\right]$

$\Updownarrow$

Subtract $1R_3$ from $R_2$

$\left[\begin{array}{ccc|c}1 & 1 & 1 & 1 \\0 & 1 & 0 & \frac{1}{3} \\ 0 & 0 & 1 & \frac{5}{3} \\\end{array}\right]$

$\Updownarrow$

Subtract $1R_3$ and $1R_2$ from $R_1$

$\left[\begin{array}{ccc|c}1 & 0 & 0 & -1 \\0 & 1 & 0 & \frac{1}{3} \\ 0 & 0 & 1 & \frac{5}{3} \\\end{array}\right]$

In [1]:
import numpy as np

A = np.matrix([[1, 1, 1], [2, 1, 1], [1, -1 ,2]])
x = np.transpose(np.matrix([[1, 0, 2]]))

print(A*x)

[[3]
 [4]
 [5]]


No $Ax \ne b$.

## Activity 2.13

Use Gaussian elimination to solve the following syhstem of equations:

$x_1+x_2+x_3=6$

$2x_1+4x_2+x_3=5$

$2x_1+3x_2+x_3=6$.

Be sure to follow the algorithm to put the augmented matrix into reduced row echelon form using row operations.

From the given linear equations we can denote $Ax=b$ as follows:

$\begin{bmatrix} 1 & 1 & 1 \\ 2 & 4 & 1 \\ 2 & 3 & 1 \end{bmatrix}\begin{bmatrix} 6 \\ 5 \\ 6 \end{bmatrix} = b$

This is also denoted as the augmented matrix:

$\left[\begin{array}{ccc|c}1 & 1 & 1 & 6 \\ 2 & 4 & 1 & 5 \\ 2 & 3 & 1 & 6 \end{array}\right]$

$\Updownarrow$

Subtract $2R_1$ from $R_2$

$\left[\begin{array}{ccc|c}1 & 1 & 1 & 6 \\ 0 & 2 & -1 & -7 \\ 2 & 3 & 1 & 6 \end{array}\right]$

$\Updownarrow$

Subtract $2R_1$ from $R_3$

$\left[\begin{array}{ccc|c}1 & 1 & 1 & 6 \\ 0 & 2 & -1 & -7 \\ 0 & 1 & -1 & -6 \end{array}\right]$

$\Updownarrow$

Multiply $R_2$ with $\frac{1}{2}$

$\left[\begin{array}{ccc|c}1 & 1 & 1 & 6 \\ 0 & 1 & -\frac{1}{2} & -\frac{7}{2} \\ 0 & 1 & -1 & -6 \end{array}\right]$

$\Updownarrow$

Subtract $1R_2$ from $R_3$

$\left[\begin{array}{ccc|c}1 & 1 & 1 & 6 \\ 0 & 1 & -\frac{1}{2} & -\frac{7}{2} \\ 0 & 0 & \frac{3}{2} & -\frac{5}{2} \end{array}\right]$

$\Updownarrow$

Add $\frac{1}{2}$ to $R_3$

$\left[\begin{array}{ccc|c}1 & 1 & 1 & 6 \\ 0 & 1 & -\frac{1}{2} & -\frac{7}{2} \\ 0 & 0 & 1 & -2 \end{array}\right]$

$\Updownarrow$

Add $\frac{1}{2}R_3$ to $R_2$

$\left[\begin{array}{ccc|c}1 & 1 & 1 & 6 \\ 0 & 1 & 0 & -\frac{11}{2} \\ 0 & 0 & 1 & -2 \end{array}\right]$

$\Updownarrow$

Subtract $1R_3$ from $R_1$

$\left[\begin{array}{ccc|c}1 & 1 & 0 & 8 \\ 0 & 1 & 0 & -\frac{11}{2} \\ 0 & 0 & 1 & -2 \end{array}\right]$

$\Updownarrow$

Subtract $1R_2$ from $R_1$

$\left[\begin{array}{ccc|c}1 & 0 & 0 & 13.5 \\ 0 & 1 & 0 & -\frac{11}{2} \\ 0 & 0 & 1 & -2 \end{array}\right]$

In [1]:
import numpy as np

A = np.matrix([[1, 1, 1], [2, 4 ,1],[2, 3, 1]])
x = np.transpose(np.matrix([6,5,6]))

np.linalg.solve(A,x)

matrix([[ 2.],
        [-1.],
        [ 5.]])

## Activity 2.15

$x_4=3$

$x_2=-1-s$

$x_1=1-t$

$$x = \begin{bmatrix} x_1 \\ x_2 \\ x_3 \\ x_4 \\ x_5 \end{bmatrix} = \begin{bmatrix} 1-t \\ -1-s \\ s \\ 3 \\ t \end{bmatrix} = \begin{bmatrix} 1 \\ -1 \\ 0 \\ 3 \\ 0 \end{bmatrix} + s \begin{bmatrix} 0 \\ -1 \\ 1 \\ 0 \\ 0 \end{bmatrix} + t \begin{bmatrix} -1 \\ 0 \\ 0 \\ 0 \\ 1 \end{bmatrix}$$

Let $s = 0$ and $t = 0$ and show (by substituting it into the equation or multiplying $Ax_0$) that $x_0= \begin{bmatrix}1 & -1 & 0 & 3 & 0 \end{bmatrix}^T$ is a solution of $Ax=b$. Then let $s=1$ and $t=2$ and show that the new vector $x_1$ you obtain is also a solution.

$s=0$ and $t=0$

$$x_0=\begin{bmatrix} 1 \\ -1 \\ 0 \\ 3 \\ 0 \end{bmatrix} + s \begin{bmatrix} 0 \\ -1 \\ 1 \\ 0 \\ 0 \end{bmatrix} + t \begin{bmatrix} -1 \\ 0 \\ 0 \\ 0 \\ 1 \end{bmatrix}$$

$$\Updownarrow$$

$$x_0=\begin{bmatrix} 1 \\ -1 \\ 0 \\ 3 \\ 0 \end{bmatrix} + 0 \begin{bmatrix} 0 \\ -1 \\ 1 \\ 0 \\ 0 \end{bmatrix} + 0 \begin{bmatrix} -1 \\ 0 \\ 0 \\ 0 \\ 1 \end{bmatrix}$$

$$\Updownarrow$$

$$x_0=\begin{bmatrix} 1 \\ -1 \\ 0 \\ 3 \\ 0 \end{bmatrix} + \begin{bmatrix} 0 \\ 0 \\ 0 \\ 0 \\ 0 \end{bmatrix} + \begin{bmatrix} 0 \\ 0 \\ 0 \\ 0 \\ 0 \end{bmatrix} = \begin{bmatrix} 1 \\ -1 \\ 0 \\ 3 \\ 0 \end{bmatrix}$$


$s=1$ and $t=2$

$$x_1 = \begin{bmatrix} 1 \\ -1 \\ 0 \\ 3 \\ 0 \end{bmatrix} + 1 \begin{bmatrix} 0 \\ -1 \\ 1 \\ 0 \\ 0 \end{bmatrix} + 2 \begin{bmatrix} -1 \\ 0 \\ 0 \\ 0 \\ 1 \end{bmatrix}$$

$$\Updownarrow$$

$$x_1 = \begin{bmatrix} 1 \\ -1 \\ 0 \\ 3 \\ 0 \end{bmatrix} +\begin{bmatrix} 0 \\ -1 \\ 1 \\ 0 \\ 0 \end{bmatrix} + \begin{bmatrix} -2 \\ 0 \\ 0 \\ 0 \\ 2 \end{bmatrix}$$

$$\Updownarrow$$

$$x_1 = \begin{bmatrix} 1 \\ -1 \\ 0 \\ 3 \\ 0 \end{bmatrix} +\begin{bmatrix} 0 \\ -1 \\ 1 \\ 0 \\ 0 \end{bmatrix} + \begin{bmatrix} -2 \\ 0 \\ 0 \\ 0 \\ 2 \end{bmatrix} = \begin{bmatrix} -1 \\ -2 \\ 1 \\ 3 \\ 2 \end{bmatrix}$$

# Exercises

## Exercise 1.4

Solve for the Matrix $A$:

$$\Big( 5A^T+\begin{bmatrix}1 & 0 \\ 2 & 5 \end{bmatrix} \Big)^T = 3A + \begin{bmatrix} 1 & -2 \\ -1 & 3 \end{bmatrix}^{-1}$$

$$\Updownarrow$$

$$5A + \begin{bmatrix} 1 & 2 \\ 0 & 5 \end{bmatrix} = 3A + \begin{bmatrix} 1 & -2 \\ -1 & 3 \end{bmatrix}^{-1}$$

$$\Updownarrow$$

$$5A + \begin{bmatrix} 1 & 2 \\ 0 & 5 \end{bmatrix} = 3A + 1 \begin{bmatrix} 1 & -2 \\ -1 & 3 \end{bmatrix}^{-1}$$

$$5A + \begin{bmatrix} 1 & 2 \\ 0 & 5 \end{bmatrix} = 3A + \begin{bmatrix} 3 & 2 \\ 1 & 1 \end{bmatrix}$$

$$\Updownarrow$$

$$2A = \begin{bmatrix} 3 & 2 \\ 1 & 1 \end{bmatrix}-\begin{bmatrix} 1 & 2 \\ 0 & 5 \end{bmatrix}$$

$$\Updownarrow$$

$$2A = \begin{bmatrix} 2 & 0 \\ 1 & -4 \end{bmatrix}$$

$$\Updownarrow$$

$$A = \begin{bmatrix} 1 & 0 \\ \frac{1}{2} & -2 \end{bmatrix}$$

## Exercise 2.2

Solve each of the following systems of equations.

1. $\Bigg\{ \begin{matrix} -x + y - 3z = 0 \\ 3x - 2y + 10z = 0 \\ -2x + 3y - 5z = 0 \end{matrix} $

$$\begin{bmatrix}-1 & 1 & -3 \\ 3 & -2 & 10 \\ -2 & 3 & -5 \end{bmatrix}$$

$$\xrightarrow{-1R_1}R_1$$

$$\begin{bmatrix} 1 & -1 & 3 \\ 3 & -2 & 10 \\ -2 & 3 & -5 \end{bmatrix}$$

$$\xrightarrow{R_2-3R_1}R_2$$
$$\xrightarrow{R_3+2R_1}R_3$$

$$\begin{bmatrix} 1 & -1 & 3 \\ 0 & 1 & 1 \\ 0 & 1 & 1 \end{bmatrix}$$

$$\xrightarrow{R_3-1R_2}$$

$$\begin{bmatrix} 1 & -1 & 3 \\ 0 & 1 & 1 \\ 0 & 0 & 0 \end{bmatrix}$$

$$\xrightarrow{R_1+1R_2}$$

$$\begin{bmatrix} 1 & 0 & 4 \\ 0 & 1 & 1 \\ 0 & 0 & 0 \end{bmatrix}$$

$$x=t\begin{bmatrix} -4 \\ -1 \\ 0 \end{bmatrix}, \space \space t \in \mathbb{R}$$

2. $\Bigg\{ \begin{matrix} -x + y - 3z = 6 \\ 3x - 2y + 10z = -10 \\ -2x + 3y - 5z = 9 \end{matrix} $

$$\left[\begin{array}{ccc|c}-1 & 1 & -3 & 6 \\ 3 & -2 & 10 \\ -2 & 3 & -5 \end{array}\right]$$

$$\xrightarrow{-1R_1}R_1$$

$$\left[\begin{array}{ccc|c}1 & -1 & 3 & -6 \\ 3 & -2 & 10 & -10 \\ -2 & 3 & -5 & 9 \end{array}\right]$$

$$\xrightarrow{R_2-3R_1}R_2$$
$$\xrightarrow{R_3+2R_1}R_3$$

$$\left[\begin{array}{ccc|c}1 & -1 & 3 & -6 \\ 0 & 1 & 1 & 8 \\ 0 & 1 & 1 & -3 \end{array}\right]$$

$$\xrightarrow{R_3-1R_2}R_3$$

$$\left[\begin{array}{ccc|c}1 & -1 & 3 & -6 \\ 0 & 1 & 1 & 8 \\ 0 & 0 & 0 & -11 \end{array}\right]$$

b is inconsistent because $0+0+0 \neq -11$

# Problems

## Problem 2.3

Express the following system of equations in matrix form:

$$-5x+y-3z-8w=3$$
$$3x+2y+2z+5w=3$$
$$x+z+2w=-1$$


$$\left[\begin{array}{cccc|c} -5 & 1 & -3 & -8 & 3 \\ 3 & 2 & 2 & 5 & 3 \\ 1 & 0 & 1 & 2 & -1 \end{array}\right]$$

Show that the system is consistent and find the general solution. Then write down the general solution of the associated homogenous system of equations.

$$\left[\begin{array}{cccc|c} -5 & 1 & -3 & -8 & 3 \\ 3 & 2 & 2 & 5 & 3 \\ 1 & 0 & 1 & 2 & -1 \end{array}\right]$$

$$\xrightarrow{R_1 \leftrightarrow R_3}$$

$$\left[\begin{array}{cccc|c} 1 & 0 & 1 & 2 & -1 \\ 3 & 2 & 2 & 5 & 3 \\ -5 & 1 & -3 & -8 & 3 \end{array}\right]$$

$$\xrightarrow{R_2-3R_1}R_2$$

$$\left[\begin{array}{cccc|c} 1 & 0 & 1 & 2 & -1 \\ 0 & 2 & -1 & -1 & 6 \\ -5 & 1 & -3 & -8 & 3 \end{array}\right]$$

$$\xrightarrow{R_3+5R_1}R_3$$

$$\left[\begin{array}{cccc|c} 1 & 0 & 1 & 2 & -1 \\ 0 & 2 & -1 & -1 & 6 \\ 0 & 1 & 2 & 2 & -2 \end{array}\right]$$

$$\xrightarrow{\frac{1}{2}R_2}$$

$$\left[\begin{array}{cccc|c} 1 & 0 & 1 & 2 & -1 \\ 0 & 1 & -\frac{1}{2} & -\frac{1}{2} & 3 \\ 0 & 1 & 2 & 2 & -2 \end{array}\right]$$

$$\xrightarrow{R_3-1R_2}R_3$$

$$\left[\begin{array}{cccc|c} 1 & 0 & 1 & 2 & -1 \\ 0 & 1 & -\frac{1}{2} & -\frac{1}{2} & 3 \\ 0 & 0 & \frac{5}{2} & \frac{5}{2} & -5 \end{array}\right]$$

$$\xrightarrow{\frac{2}{5}R_3}R_3$$

$$\left[\begin{array}{cccc|c} 1 & 0 & 1 & 2 & -1 \\ 0 & 1 & -\frac{1}{2} & -\frac{1}{2} & 3 \\ 0 & 0 & 1 & 1 & -2 \end{array}\right]$$

$$\xrightarrow{R_1-1R_3}R_1$$

$$\left[\begin{array}{cccc|c} 1 & 0 & 0 & 1 & 1 \\ 0 & 1 & -\frac{1}{2} & -\frac{1}{2} & 3 \\ 0 & 0 & 1 & 1 & -2 \end{array}\right]$$

$$\xrightarrow{R_2-\frac{1}{2}R_3}$$

$$\left[\begin{array}{cccc|c} 1 & 0 & 0 & 1 & 1 \\ 0 & 1 & 0 & 0 & 4 \\ 0 & 0 & 1 & 1 & -2 \end{array}\right]$$

$$x = t \begin{bmatrix} -1 \\ 0 \\ -1 \end{bmatrix}, \space t \in \mathbb{R}$$

## Problem 2.9

$$R=\begin{bmatrix} 1.05 & 0.95 & 1.0 \\ 1.05 & 1.05 & 1.05 \\ 1.20 & 1.26 & 1.23 \end{bmatrix}$$

1. $W=\begin{bmatrix} 5000 & 2000 & 0 \end{bmatrix}$

$$WR = W=\begin{bmatrix} 5000 & 2000 & 0 \end{bmatrix}\begin{bmatrix} 1.05 & 0.95 & 1.0 \\ 1.05 & 1.05 & 1.05 \\ 1.20 & 1.26 & 1.23 \end{bmatrix}$$

In [29]:
import numpy as np

W = np.matrix([5000, 2000, 0])
R = np.matrix([[1.05, 0.95, 1.0], [1.05, 1.05, 1.05], [1.20, 1.26, 1.23]])

print(W*R)

[[7350. 6850. 7100.]]


$$WR = \begin{bmatrix}7350 & 6850 & 7100 \end{bmatrix}$$

2. $U=\begin{bmatrix} 600 & 8000 & 1000 \end{bmatrix}$

$$UR = U=\begin{bmatrix} 600 & 8000 & 1000 \end{bmatrix}\begin{bmatrix} 1.05 & 0.95 & 1.0 \\ 1.05 & 1.05 & 1.05 \\ 1.20 & 1.26 & 1.23 \end{bmatrix}$$

In [30]:
U = np.matrix([600, 8000, 1000])

print(U*R)

[[10230. 10230. 10230.]]


$$UR = \begin{bmatrix} 10230 & 10230 & 10230 \end{bmatrix}$$

3. $Z=\begin{bmatrix} 1000 & -2000 & 1000 \end{bmatrix}$

$$ZR = \begin{bmatrix} 1000 & -2000 & 1000 \end{bmatrix}\begin{bmatrix} 1.05 & 0.95 & 1.0 \\ 1.05 & 1.05 & 1.05 \\ 1.20 & 1.26 & 1.23 \end{bmatrix}$$

In [31]:
Z = np.matrix([[1000, -2000, 1000]])

print(Z*R)

[[150. 110. 130.]]


$$ZR = \begin{bmatrix} 150 & 110 & 130 \end{bmatrix}$$